### PR TITLE
test: card

### PR DESCRIPTION
**Pull request by SzDabrowski**

### CODE REVIEW

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
## Code Review Report

### Task Goal
The task goal was to create unit tests for the `<Card />` component. The requirements were to test if:

1. Clicking on `<Card />` enters edit mode.
2. Clicking outside edit mode exits edit mode.
3. `onUpdateCard` is being fired when exiting edit mode.
4. The value on textarea is being set when firing change event in edit mode.

### Code Changes Review

#### db.json
The changes in `db.json` are not relevant to the task requirements. The task requires creating unit tests for the `<Card />` component, and does not involve any changes to the database.

#### src/components/Card/Card.test.tsx
The changes in `src/components/Card/Card.test.tsx` cover all the task requirements:

1. The test 'enters edit mode on click' checks if clicking on `<Card />` enters edit mode.
2. The test 'exits edit mode on click outside' checks if clicking outside edit mode exits edit mode.
3. The test 'fires onUpdateCard when exiting edit mode' checks if `onUpdateCard` is being fired when exiting edit mode.
4. The test 'sets textarea value on change in edit mode' checks if the value on textarea is being set when firing change event in edit mode.

However, there's a minor issue in the code. The `vi.fn()` function is used to create a mock function, but it's not a standard Jest function. It should be replaced with `jest.fn()`.

Here's the corrected code:

```diff
- const mockUpdateCard = vi.fn()
+ const mockUpdateCard = jest.fn()
```

Overall, the code changes cover all the task requirements and the tests are well written. They cover all the necessary user interactions and check the correct component responses.

**Review by NerdbordMentor — APPROVED** [review]
Based on the provided task requirements and code changes, the changes seem to cover all the requirements adequately. Here's a breakdown:

1. **Test if clicking on `<Card />` enters edit mode**: The test 'enters edit mode on click' covers this requirement. It simulates a click event on the card and checks if a textarea (assumed to be the edit mode UI element) is present in the document.

2. **Test if clicking outside edit mode exits edit mode**: The test 'exits edit mode on click outside' covers this requirement. It simulates entering and exiting the edit mode by firing click and blur events respectively, and checks if the textarea is not present in the document after the blur event.

3. **Test if `onUpdateCard` is being fired when exiting edit mode**: The test 'fires onUpdateCard when exiting edit mode' covers this requirement. It renders the Card component with a mock `onUpdateCard` function, simulates entering and exiting the edit mode, and checks if the mock function was called.

4. **Test if value on textarea is being set when firing change event in edit mode**: The test 'sets textarea value on change in edit mode' covers this requirement. It simulates entering the edit mode and changing the value of the textarea, and checks if the new value is set correctly.

The changes in `db.json` don't seem to be directly related to the task requirements, but they don't conflict with them either.

In conclusion, the code changes appear to meet all the task requirements.
